apps/discovery_streams/lib/discovery-streams/stream/source_handler.ex

```diff
@@ -0,0 +1,80 @@
+defmodule DiscoveryStreams.Stream.SourceHandler do
+  @moduledoc """
+  Callbacks for handling data messages.
+  See [Source.Handler](../../../../protocol_source/lib/source/handler.ex)
+  for more.
+  """
+  use Source.Handler
+  use Properties, otp_app: :discovery_streams
+  require Logger
+  alias StreamingMetrics.Hostname
+
+  alias DiscoveryStreamsWeb.Endpoint
+
+  getter(:dlq, default: Dlq)
+
+  def handle_message(message, context) do
+    dataset_id = context.dataset_id
+
+    Logger.debug(fn -> "#{__MODULE__} handle_message - #{inspect(context)} - #{inspect(message)}" end)
+
+    log_message(message)
+
+    case Brook.get(:discovery_streams, :streaming_datasets_by_id, dataset_id) do
+      {:ok, system_name} ->
+        Endpoint.broadcast!("streaming:#{system_name}", "update", message)
+
+      _ ->
+        nil
+    end
+
+    Ok.ok(message)
+  end
+
+  def handle_batch(batch, context) do
+    Logger.debug(fn -> "#{__MODULE__} handle_batch - #{inspect(context)} - #{inspect(batch)}" end)
+    record_outbound_count_metrics(batch, context.dataset_id)
+    :ok
+  end
+
+  defp log_message(message) do
+    Logger.log(:info, "#{inspect(message)}")
+    message
+  end
+
+  def send_to_dlq(dead_letters, _context) do
+    dlq().write(dead_letters)
+    :ok
+  end
+
+  defp record_outbound_count_metrics(messages, dataset_id) do
+    messages
+    |> Enum.reduce(%{}, fn _, acc -> Map.update(acc, dataset_id, 1, &(&1 + 1)) end)
+    |> Enum.each(&record_metric/1)
+  end
+
+  defp record_metric({dataset_id, count}) do
+    pretty_dataset_id =
```

Comment:
Was telemetry complaining about this? Life's easier in metrics land when metrics have the same labels and I believe the other apps don't do this

Reply:
You mean the  - to _ conversion?  That code was just already there for topics so i assumed someone was picky, I can nix this if you want.